cli_tools/common/utils/paramhelper/network.go

```diff
@@ -0,0 +1,62 @@
+//  Copyright 2021 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package paramhelper
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+)
+
+var (
+	// Definition of resource path:
+	//   https://cloud.google.com/compute/docs/reference/rest/v1/networks/get
+	networkPattern = regexp.MustCompile(`^(?:projects/([^/]+)/)?(?:global/networks/)?([^/]+)$`)
+)
+
+// NetworkResource contains the name and project of a compute.Network resource.
+type NetworkResource struct {
+	Name, Project string
+}
+
+// String formats the NetworkResource as a GCP-style resource identifier.
+func (r *NetworkResource) String() string {
+	if r.Name == "" {
+		return ""
+	}
+	if r.Project == "" {
+		return fmt.Sprintf("global/networks/%s", r.Name)
+	}
+	return fmt.Sprintf("projects/%s/global/networks/%s", r.Project, r.Name)
```

Comment:
minor: slight duplication here, up to you if you want to avoid it.

Reply:
updated